bitclaw/ecommerce-assessment
# Feature/initial setup #1

* Update README.md
* Run `bun install` to install dependencies and commit generated lock file
* Configure Tailwind (postcss.config.js and global.css)
* Add initial nav component
* Add home page (page.tsx)
* Add favicon
* Add font that we are going to use
* Add provided images for product
* Add NextJS configuration for security headers
* Add initial setup script, I may use it later to populate data on local machine for testing purposes
* Add initial layout just to get the site running
* Run `bun dev` to verify basic layout and nav bar is working
* Commit tsconfig.json generated when the command `bun dev` was run for the first time
* When strictNullChecks is false , null and undefined are effectively ignored by the language. This can lead to unexpected errors at runtime.
* Ignore .env file, it should not be committed to the repo since it may contain sensitive credentials
* Remove unneeded comment
* Forgot to add tailwind configuration to get styles to render
* Add additional dependencies for testing
* Add new test script to run vitest for React tests
* Move dev dependencies in the package.json, so we know what dependencies are only used for local development
* Update README.md
* Configure vitest and add a component test and a server function test
* Update README.md with correct command to run tests
* Fix failing test
